functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/LogViewerOutputSpec.groovy

```diff
@@ -48,45 +48,30 @@ class LogViewerOutputSpec extends SeleniumBase{
 
     def "click on gutter and refresh should highlight correct line"() {
 
-        setup:
+        given:
         def loginPage = page LoginPage
         def sideBar = page SideBarPage
-        def jobListPage = page JobsListPage
-        def jobCreatePage = page JobCreatePage
         def projectHomePage = page HomePage
         def jobShowPage = page JobShowPage
 
-        when:
+        when: "We run the job to have multiple lines in log output"
         loginPage.go()
         loginPage.login(TEST_USER, TEST_PASS)
         projectHomePage.validatePage()
         projectHomePage.goProjectHome(longOutPutProjectName)
         sideBar.goTo(NavLinkTypes.JOBS)
-        jobListPage.getCreateJobLink().click()
-        jobCreatePage.getJobNameField().sendKeys("loop job")
-        jobCreatePage.getTab(JobTab.WORKFLOW).click()
-        jobCreatePage.getStepByType(StepName.COMMAND, StepType.NODE).click()
-        jobCreatePage.waitForStepToBeShown(By.name("pluginConfig.adhocRemoteString"))
-        jobCreatePage.el(By.name("pluginConfig.adhocRemoteString")).sendKeys("for i in {1..6}; do echo NUMBER \$i; sleep 0.5; done")
-        jobCreatePage.getSaveStepButton().click()
-        jobCreatePage.waitForSavedStep(0)
-        jobCreatePage.getCreateButton().click()
+        jobShowPage.goToJob("f44481c4-159d-4176-869b-e4a9bd898fe4")
         jobShowPage.getRunJobBtn().click()
         jobShowPage.getLogOutputBtn().click()
-        jobShowPage.waitForLogOutput(By.xpath("//span[contains(text(),'NUMBER ')]"),3,5)
-        def lineToClick = jobShowPage.el(By.xpath("//span[contains(text(),'NUMBER 1')]/ancestor::div[contains(@class, 'execution-log__line')]/div[@class='execution-log__gutter']"))
-
+        jobShowPage.waitForLogOutput(By.xpath("//span[contains(text(),'test output ')]"),9,5)
+        def lineToClick = jobShowPage.el(By.xpath("//span[contains(text(),'test output 5')]/ancestor::div[contains(@class, 'execution-log__line')]/div[@class='execution-log__gutter']"))
         lineToClick.click();
-        jobShowPage.waitForUrlToContain("#outputL1")
+        jobShowPage.waitForUrlToContain("#outputL5")
         driver.navigate().refresh();
-//        def href = commandPage.runningButtonLink().getAttribute("href")
-//        jobShowPage.driver.get href + "#outputL1"
-
-        jobShowPage.waitForLogOutput(By.xpath("//span[contains(text(),'NUMBER ')]"),3,5)
 
         then:
+        jobShowPage.waitForUrlToContain("#outputL5")
         def selectedLine = jobShowPage.waitForElementVisible(By.xpath("//div[contains(@class, 'execution-log__line--selected')]"))
-
         assert selectedLine != null : "Expected at least one element with the specified class to be present"
```

Comment:
The 'assert' keyword is unnecessary here, since "then" evaluates all lines as expressions.